vendor/bat-native-ledger/src/bat/ledger/internal/legacy/client_properties.cc

```diff
@@ -47,4 +68,133 @@ bool ClientProperties::operator!=(
   return !(*this == rhs);
 }
 
+base::Value::Dict ClientProperties::ToValue() const {
+  base::Value::Dict dict;
+  dict.Set(kWalletInfoKey, wallet_info.ToValue());
+  dict.Set(kBootTimestampKey, base::Int64ToValue(boot_timestamp));
+  dict.Set(kReconcileTimestampKey, base::Int64ToValue(reconcile_timestamp));
+  dict.Set(kFeeAmountKey, fee_amount);
+  dict.Set(kUserChangedFeeKey, user_changed_fee);
+  dict.Set(kRewardsEnabledKey, rewards_enabled);
+  dict.Set(kAutoContributeKey, auto_contribute);
+
+  base::Value::Dict inline_tips_dict;
+  for (const auto& [key, value] : inline_tips) {
+    inline_tips_dict.Set(key, value);
+  }
+  dict.Set(kInlineTipsKey, std::move(inline_tips_dict));
+
+  return dict;
+}
+
+bool ClientProperties::FromValue(const base::Value::Dict& dict) {
+  // Wallet Info
+  const auto* wallet_info_dict = dict.FindDict(kWalletInfoKey);
+  if (!wallet_info_dict) {
+    NOTREACHED();
+    return false;
+  }
+  if (!wallet_info.FromValue(*wallet_info_dict)) {
+    NOTREACHED();
+    return false;
+  }
+
+  // Boot Timestamp (There is no support for uint64_t. Writing JSON with such
+  // types violates the spec. As we need a uint64_t, we need to use an unsigned
+  // int and cast to a uint64_t)
+  if (auto value = base::ValueToInt64(dict.Find(kBootTimestampKey))) {
+    boot_timestamp = *value;
+  } else if (auto value = dict.FindDouble(kBootTimestampKey)) {
+    boot_timestamp = static_cast<uint64_t>(*value);
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Reconcile Timestamp (There is no support for uint64_t. Writing JSON with
+  // such types violates the spec. As we need a uint64_t, we need to use an
+  // unsigned int and cast to a uint64_t)
+  if (auto value = base::ValueToInt64(dict.Find(kReconcileTimestampKey))) {
+    reconcile_timestamp = *value;
+  } else if (auto value = dict.FindDouble(kReconcileTimestampKey)) {
+    reconcile_timestamp = static_cast<uint64_t>(*value);
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Fee Amount
+  if (auto value = dict.FindDouble(kFeeAmountKey)) {
+    fee_amount = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // User Changed Fee
+  if (auto value = dict.FindBool(kUserChangedFeeKey)) {
+    user_changed_fee = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Auto Contribute
+  if (auto value = dict.FindBool(kAutoContributeKey)) {
+    auto_contribute = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Rewards Enabled
+  if (auto value = dict.FindBool(kRewardsEnabledKey)) {
+    rewards_enabled = *value;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  // Inline Tips
+  if (const auto* value = dict.FindDict(kInlineTipsKey)) {
+    for (const auto inline_tip_value : *value) {
```

Comment:
This is copying the `std::pair<>` of references. You can use a structured binding here and `emplace<>()` on the map.